noxfile.py

```diff
@@ -136,13 +142,25 @@ def tests(session, shard):
     if shard == "llamaindex":
         session.posargs.insert(0, "-n4")
 
-    session.run(
+    # Add sharding logic for trace1, trace2, trace3
+    pytest_args = [
         "pytest",
         "--durations=20",
         "--strict-markers",
         "--cov=weave",
         "--cov-report=html",
         "--cov-branch",
+    ]
+
+    # Handle trace sharding: run every 3rd test starting at different offsets
+    if shard in trace_shards:
+        shard_id = int(shard[-1]) - 1
```

Comment:
Why not just index the shard names from 0 so you don't have to subtract here and add in the list comprehension above?